tests/functional/object/rgw/test_rgw_http_endpoint.py

```
@@ -0,0 +1,298 @@
+
+        # Step 2: Validate unsecured RGW route is present in route info
+        log.info("Step 2: Validating unsecured RGW route is present")
+        http_route_data = self.wait_for_route_to_exist(
+            constants.RGW_ROUTE_INTERNAL_MODE, timeout=180
+        )
+        log.info("HTTP route exists as expected when disableHttp is False")
+
+        # Save the HTTP endpoint URL for later testing in Step 5
+        http_host = http_route_data.get("spec", {}).get("host")
+        http_endpoint_url = f"http://{http_host}"
+        log.info(f"Saved HTTP endpoint URL: {http_endpoint_url}")
```

Ruff (0.15.14)

[warning] Logging statement uses f-string

(G004)